python/lsst/daf/butler/core/formatter.py

```diff
@@ -210,6 +210,27 @@ def write(self, inMemoryDataset: Any) -> str:
         """
         raise NotImplementedError("Type does not support writing")
 
+    @classmethod
+    def can_read_bytes(cls) -> bool:
+        """Indicate if this formatter can format from bytes.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the `fromBytes` method is implemented.
+        """
+        # We have no property to read so instead try to format from a byte
+        # and see what happens
+        try:
+            # We know the arguments are incompatible
+            cls.fromBytes(cls, b"")  # type: ignore
+        except NotImplementedError:
+            return False
+        except Exception:
+            # There will be problems with the bytes we are supplying so ignore
+            pass
+        return True
+
```

Comment:
Is the long-term plan here to implement this in subclasses directly, and then make it abstract?

Reply:
Yes. I haven't worked out the best approach yet but there is a Formatter cleanup ticket. I wanted an API here though so that I could at least ask the formatter about it even if the implementation was a bit odd. I had hoped to get away with always using the presence of a method to declare it but it seems like I'm going to have to use the approach of a class property and implementation method.